Rdmp.Core/Curation/Data/ImportExport/ShareManager.cs

```diff
@@ -331,13 +333,28 @@ public void DeleteAllOrphanImportDefinitions()
                 var instance = (IMapsDirectlyToDatabaseTable)ObjectConstructor.ConstructIfPossible(sd.Type, this, sd) ??
                                throw new ObjectLacksCompatibleConstructorException(
                                    $"Could not find a ShareManager constructor for '{sd.Type}'");
+                if (instance.GetType() == typeof(LoadMetadataCatalogueLinkage))
+                {
+                    //find most recent lmd and all catalogues since, then link them
+                    var latestLMD = created.OfType<LoadMetadata>().Last();
```

Comment:
`.Last()` throws an exception on empty lists rather than returning `null` - should this be `LastOrDefault()` for the next line checking for null to make sense?